object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/CalledMethodsPredicateEvaluator.java

```diff
@@ -1,23 +1,181 @@
 package org.checkerframework.checker.objectconstruction;
 
+import com.github.javaparser.JavaParser;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.expr.BinaryExpr;
+import com.github.javaparser.ast.expr.UnaryExpr;
+import com.github.javaparser.ast.stmt.BlockStmt;
 import java.util.Collection;
+import org.checkerframework.javacutil.BugInCF;
+import org.sosy_lab.common.ShutdownManager;
+import org.sosy_lab.common.ShutdownNotifier;
+import org.sosy_lab.common.configuration.Configuration;
+import org.sosy_lab.common.configuration.InvalidConfigurationException;
+import org.sosy_lab.common.log.LogManager;
+import org.sosy_lab.java_smt.SolverContextFactory;
+import org.sosy_lab.java_smt.api.BooleanFormula;
+import org.sosy_lab.java_smt.api.BooleanFormulaManager;
+import org.sosy_lab.java_smt.api.ProverEnvironment;
+import org.sosy_lab.java_smt.api.SolverContext;
 import org.springframework.expression.Expression;
 import org.springframework.expression.ExpressionParser;
 import org.springframework.expression.spel.standard.SpelExpressionParser;
 
-/** This class parses and evaluates a single @CalledMethodsPredicate argument. */
+/**
+ * This class contains static methods that parse and evaluate the arguments of {@link
+ * org.checkerframework.checker.objectconstruction.qual.CalledMethodsPredicate} annotations, so that
+ * they can be compared to each other and to {@link
+ * org.checkerframework.checker.objectconstruction.qual.CalledMethods} annotations.
+ */
 public class CalledMethodsPredicateEvaluator {
 
-  // All the names of methods that ought to evaluate to true.
-  private final Collection<String> cmMethods;
+  /** The parser to use when converting formulae to ASTs. */
+  private static final JavaParser parser = new JavaParser();
 
-  public CalledMethodsPredicateEvaluator(final Collection<String> cmMethods) {
-    this.cmMethods = cmMethods;
+  /** Do not use. This class is not instantiable; use the static methods instead. */
+  private CalledMethodsPredicateEvaluator() {
+    throw new Error("Do not instantiate");
   }
 
-  protected boolean evaluate(String expression) {
+  /** Construct a solver to use when comparing predicates. Each query should use a new solver. */
+  private static SolverContext setupSolver() {
+    Configuration config = Configuration.defaultConfiguration();
+    LogManager log = LogManager.createNullLogManager();
+    ShutdownNotifier notifier = ShutdownManager.create().getNotifier();
+    try {
+      return SolverContextFactory.createSolverContext(
+          config, log, notifier, SolverContextFactory.Solvers.SMTINTERPOL);
+    } catch (InvalidConfigurationException e) {
+      return null;
```

Comment:
This is guaranteed to lead to a null pointer exception later, which will be harder to debug.
More generally, the code suppresses a number of exceptions and I think it would be better to at least log them, to help users understand unexpected behavior.